TC-IDM-5.2: Fix status code in test and SDK

cecille opened the pull request:
Please see:
https://github.com/project-chip/connectedhomeip/issues/33936

cecille committed:
TC-IDM-5.2: Fix status code in test and SDK

Please see:
https://github.com/project-chip/connectedhomeip/issues/33936

bzbarsky-apple commented on src/app/TimedHandler.cpp:
```diff
@@ -66,10 +66,9 @@ CHIP_ERROR TimedHandler::OnMessageReceived(Messaging::ExchangeContext * aExchang
             ChipLogError(DataManagement, "Timeout expired: handler %p exchange " ChipLogFormatExchange, this,
                          ChipLogValueExchange(aExchangeContext));
-            return StatusResponse::Send(Status::UnsupportedAccess, aExchangeContext, /* aExpectResponse = */ false);
+            return StatusResponse::Send(Status::Timeout, aExchangeContext, /* aExpectResponse = */ false);
```
Looking at https://github.com/CHIP-Specifications/connectedhomeip-spec/commit/0bd0e4f937ebaf84d6ebe7dc22e79c5af605bc2a and the current spec text we also need to fix `WriteHandler::ProcessWriteRequest`.  The `mIsTimedRequest != aIsTimedWrite` bit needs to return TimedRequestMismatch, right?  It looks like this was changed for invoke but not write, probably because we don't have any actual timed writable attributes to test on....

cecille replied:
If it's ok, I'd like to address this in a followup. We can do a non-cert test of a writable timed attribute using the unit testing cluster, but I don't believe I will have a change to add attributes and tests etc. before the TE2 deadline. I can make an issue if the write is not a blocker for this PR.

cecille replied:
As it turns out, we have a timed write attribute in the unit testing cluster and a test, so may as well do this now.

cecille replied:
Actually, it turns out we DO have a unit test for this, so I'm more comfortable just fixing this here. Opened an issue to add to cert as that's going to require a LOT more plumbing. Issue here: https://github.com/CHIP-Specifications/chip-test-plans/issues/4256

cecille committed:
Fix unit tests

We have unit tests! They didn't match the spec, but they exist!

One for two is still better than 0 for 2.

cecille committed:
Fix unit test

cecille committed:
Fix error on write request with incorrect timed request flag

Please see
https://github.com/project-chip/connectedhomeip/pull/33952#discussion_r1643204412

This was changed in the spec, but not in the SDK or unit tests.

The unit test here covers this, but we should consider adding it to
the cert testing as well. Please see
https://github.com/CHIP-Specifications/chip-test-plans/issues/4256

bzbarsky-apple reviewed (approved):
Thank you!

cecille committed:
fix write in test as well
